### PR TITLE
feat: updating frontend enterprise to include updated program filters

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
[Jira](https://openedx.atlassian.net/browse/ENT-4388)
screenshots:
image

mobile:
image

Work done:
Bumping enterprise frontend version

Before this is merged:
Merge [the edx internal pr](asheehan-edx/ENT-4388-update-program-filter-feature-flags) to add the feature flag

To test update your .env.development file to
```
FEATURE_PROGRAM_TITLES_FACET=true
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "@edx/brand": "npm:@edx/brand-edx.org@2.0.0",
     "@edx/frontend-component-footer": "10.1.4",
     "@edx/frontend-component-header": "2.2.4",
-    "@edx/frontend-enterprise": "5.3.3",
+    "@edx/frontend-enterprise": "^5.5.1",
```
Comment (binodpant) [inline package.json]: I have started to customarily / habitually remove the caret here (and re run npm i if you do that), a bot will eventually clear this up but don't know how often/diligently we merge those bot PRs

Comment (adamstankiewicz) [reply package.json]: Not often/diligently haha. I've gotten into that same habit ;)

**Review by raq929 — APPROVED** [review]

File: .env.development [hunk 2]
```diff
@@ -28,3 +28,5 @@ MINIMAL_HEADER=true
 EDX_FOR_BUSINESS_UUID=''
 EDX_FOR_ONLINE_EDU_UUID=''
 EDX_ONLINE_ESSENTIALS_UUID=''
+FEATURE_PROGRAM_TITLES_FACET=null
```
Comment (raq929) [inline .env.development]: Wait, why are these set to null in .env.development? Shouldn't they be set to true?

Comment (alex-sheehan-edx) [reply .env.development]: gooood point, my thinking was to leave it up to whoever's developing to turn the features on or off but I guess there's really no reason to have it off. I'll change the default to True!

**Review by raq929 — APPROVED** [review]

**Comment by davidjoy** [comment]
tiny nit: may want to have the prefix be "feat" rather than "FEAT" - I think the latter may not technically conform to conventional commits. :)

**Comment by alex-sheehan-edx** [comment]
jenkins run python

**Pull request merged**